lib/cisco_node_utils/interface.rb

```diff
@@ -303,6 +303,39 @@ def default_negotiate_auto
       config_get_default('interface', negotiate_auto_lookup_string)
     end
 
+    def port_channel
+      config_get('interface', 'port_channel', @name)
+    end
+
+    def port_channel=(val)
+      case @name
+      when /Ethernet/i
```

Comment:
I don't see a need for the `case` if there's only ever one interface type that's allowed. Just do:

```
fail "channel_group is not supported on #{name}" unless @name[/Ethernet/i]
```